### DISCUSSION TITLE
No PDS text in tornado watches via API

### DISCUSSION THREAD

**Discussion by JJKraw** [weather-gov/api#223] [Q&A]
Given that Particularly Dangerous Situation Tornado Watches have been issued today, I'm not finding that text in the watches via the alerts endpoints in the API.  For example, here is watch 29 text from the API:

https://api.weather.gov/alerts/urn:oid:2.49.0.1.840.0.f8ba5f781e0fc11aff39eb81a383f2366818ef69.001.1

***
THE NATIONAL WEATHER SERVICE HAS ISSUED TORNADO WATCH 29 IN
EFFECT UNTIL 7 PM CDT THIS EVENING FOR THE FOLLOWING AREAS

IN ALABAMA THIS WATCH INCLUDES 23 COUNTIES

IN CENTRAL ALABAMA

AUTAUGA               BIBB                  BLOUNT
CHILTON               COOSA                 DALLAS
JEFFERSON             PERRY                 SHELBY
ST. CLAIR             TALLADEGA             WALKER

IN EAST CENTRAL ALABAMA

CLAY

IN NORTHWEST ALABAMA

MARION                WINSTON

IN WEST CENTRAL ALABAMA

FAYETTE               GREENE                HALE
LAMAR                 MARENGO               PICKENS
SUMTER                TUSCALOOSA

THIS INCLUDES THE CITIES OF ALABASTER, ALICEVILLE, ASHLAND,
BIRMINGHAM, BRENT, CARROLLTON, CENTREVILLE, CLANTON, COLUMBIANA,
DEMOPOLIS, EUTAW, FAYETTE, GREENSBORO, HALEYVILLE, HAMILTON,
HOOVER, JASPER, LINDEN, LINEVILLE, LIVINGSTON, MARION, MOODY,
MOUNDVILLE, ONEONTA, PELHAM, PELL CITY, PRATTVILLE, ROCKFORD,
SELMA, SULLIGENT, SYLACAUGA, TALLADEGA, TUSCALOOSA, UNIONTOWN,
VERNON, AND YORK.
***

The text directly from the SPC site is more detailed and does contain the phrasing.

https://www.spc.noaa.gov/products/watch/ww0029.html

***
URGENT - IMMEDIATE BROADCAST REQUESTED
   Tornado Watch Number 29
   NWS Storm Prediction Center Norman OK
   1135 AM CDT Wed Mar 17 2021

   The NWS Storm Prediction Center has issued a

   * Tornado Watch for portions of 
     Western and central Alabama
     Central to eastern Mississippi

   * Effective this Wednesday morning and evening from 1135 AM until
     700 PM CDT.

   ...THIS IS A PARTICULARLY DANGEROUS SITUATION...

   * Primary threats include...
     Numerous tornadoes expected with a few intense tornadoes likely
     Scattered damaging winds and isolated significant gusts to 80
       mph likely
     Scattered large hail and isolated very large hail events to 3
       inches in diameter likely

   SUMMARY...Numerous discrete supercells are expected to develop this
   afternoon capable of producing strong to intense tornadoes.

   The tornado watch area is approximately along and 80 statute miles
   east and west of a line from 70 miles northwest of Gadsden AL to 35
   miles southeast of Pine Belt MS. For a complete depiction of the
   watch see the associated watch outline update (WOUS64 KWNS WOU9).

   PRECAUTIONARY/PREPAREDNESS ACTIONS...

   REMEMBER...A Tornado Watch means conditions are favorable for
   tornadoes and severe thunderstorms in and close to the watch
   area. Persons in these areas should be on the lookout for
   threatening weather conditions and listen for later statements
   and possible warnings.

   &&

   OTHER WATCH INFORMATION...CONTINUE...WW 26...WW 27...WW 28...

   AVIATION...Tornadoes and a few severe thunderstorms with hail
   surface and aloft to 3 inches. Extreme turbulence and surface wind
   gusts to 70 knots. A few cumulonimbi with maximum tops to 500. Mean
   storm motion vector 24030.

   ...Grams
***

It would seem that containing the PDS phrase in the alerts from the API should be something that is desired. Is there something in the transfer of information from the SPC to CAP alerts (I notice alerts.weather.gov doesn't have the PDS phrase either) that prevents this?

Hope this all makes sense.  Thanks.

**Comment by StephenClouse** [comment]
Our group doesn't build the CAP, we receive it from upstream. This is something we'll have to pass along to those folks.

At a glance it looks like they build the CAP from the WOU (watch outline) and not the SEL (the actual watch product). Which makes sense, because only the WOU contains the metadata for which counties are in the watch.

**Comment by JJKraw** [comment]
Thank you. So if I understand correctly, you'll pass the info along. Let me know if I should contact anyone else.

I assume the NWS really wants the PDS info to be there in the end products.

Thanks again!

**Comment by StephenClouse** [comment]
Right, API simply syndicates the CAP feed as distributed by NCO IDP. We do not modify the data (other than serializing it to JSON). So any change would need to come from there.

We work pretty closely with that team so I'll see what they think.

**Comment by scadergit** [comment]
In general, any improvements to NWS products or services outside of the performance of the software can be directed to your local weather office. They take requests seriously, and have established protocols for raising them.
